
### Tugas W2S2
42324001 - Oktova Samosir

In [1]:
import pandas as pd
path = "shopping_trends.csv"
df = pd.read_csv(path)

df

,Customer ID,Age,Gender,Item Purchased,Category,Purchase Amount (USD),Location,Size,Color,Season,Review Rating,Subscription Status,Payment Method,Shipping Type,Discount Applied,Promo Code Used,Previous Purchases,Preferred Payment Method,Frequency of Purchases
0,1,55,Male,Blouse,Clothing,53,Kentucky,L,Gray,Winter,3.1,Yes,Credit Card,Express,Yes,Yes,14,Venmo,Fortnightly
1,2,19,Male,Sweater,Clothing,64,Maine,L,Maroon,Winter,3.1,Yes,Bank Transfer,Express,Yes,Yes,2,Cash,Fortnightly
2,3,50,Male,Jeans,Clothing,73,Massachusetts,S,Maroon,Spring,3.1,Yes,Cash,Free Shipping,Yes,Yes,23,Credit Card,Weekly
3,4,21,Male,Sandals,Footwear,90,Rhode Island,M,Maroon,Spring,3.5,Yes,PayPal,Next Day Air,Yes,Yes,49,PayPal,Weekly
4,5,45,Male,Blouse,Clothing,49,Oregon,M,Turquoise,Spring,2.7,Yes,Cash,Free Shipping,Yes,Yes,31,PayPal,Annually
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
3895,3896,40,Female,Hoodie,Clothing,28,Virginia,L,Turquoise,Summer,4.2,No,Cash,2-Day Shipping,No,No,32,Venmo,Weekly
3896,3897,52,Female,Backpack,Accessories,49,Iowa,L,White,Spring,4.5,No,PayPal,Store Pickup,No,No,41,Bank Transfer,Bi-Weekly
3897,3898,46,Female,Belt,Accessories,33,New Jersey,L,Green,Spring,2.9,No,Credit Card,Standard,No,No,24,Venmo,Quarterly
3898,3899,44,Female,Shoes,Footwear,77,Minnesota,S,Brown,Summer,3.8,No,PayPal,Express,No,No,24,Venmo,Weekly


# Analisis Customer Shopping Trends

Pada tugas ini, kita akan melakukan analisis terhadap dataset
Customer Shopping Trends.

Analisis yang dilakukan meliputi:
1. Analisis statistik pada kolom `Previous Purchases`.
2. Analisis korelasi `Age` dengan `Purchase Amount (USD)`.
3. Analisis korelasi `Purchase Amount (USD)` dengan `Review Rating`.

Library Pandas digunakan untuk membaca dan mengolah dataset.

## Import Dataset

Dataset Customer Shopping Trends disimpan dalam file CSV.

Pada tahap ini, file CSV dibaca menggunakan fungsi `pd.read_csv()`.
Variabel `path` digunakan untuk menentukan nama atau lokasi file dataset.

Data yang telah dibaca kemudian disimpan ke dalam variabel `df`
dalam bentuk DataFrame.

In [21]:
df.head()

,Customer ID,Age,Gender,Item Purchased,Category,Purchase Amount (USD),Location,Size,Color,Season,Review Rating,Subscription Status,Payment Method,Shipping Type,Discount Applied,Promo Code Used,Previous Purchases,Preferred Payment Method,Frequency of Purchases
0,1,55,Male,Blouse,Clothing,53,Kentucky,L,Gray,Winter,3.1,Yes,Credit Card,Express,Yes,Yes,14,Venmo,Fortnightly
1,2,19,Male,Sweater,Clothing,64,Maine,L,Maroon,Winter,3.1,Yes,Bank Transfer,Express,Yes,Yes,2,Cash,Fortnightly
2,3,50,Male,Jeans,Clothing,73,Massachusetts,S,Maroon,Spring,3.1,Yes,Cash,Free Shipping,Yes,Yes,23,Credit Card,Weekly
3,4,21,Male,Sandals,Footwear,90,Rhode Island,M,Maroon,Spring,3.5,Yes,PayPal,Next Day Air,Yes,Yes,49,PayPal,Weekly
4,5,45,Male,Blouse,Clothing,49,Oregon,M,Turquoise,Spring,2.7,Yes,Cash,Free Shipping,Yes,Yes,31,PayPal,Annually


## Melihat Lima Data Pertama

Fungsi `head()` digunakan untuk melihat lima baris pertama dari
DataFrame.

Tahap ini dilakukan untuk mengetahui gambaran awal mengenai isi
dataset, seperti nama kolom dan contoh nilai yang terdapat di
dalamnya.

## Informasi Dataset

Fungsi `info()` digunakan untuk melihat informasi umum mengenai
dataset.

Informasi yang ditampilkan meliputi:
- jumlah baris dan kolom,
- nama setiap kolom,
- tipe data setiap kolom,
- jumlah data yang tidak kosong pada setiap kolom.

Informasi ini berguna untuk mengetahui struktur dataset sebelum
melakukan analisis lebih lanjut.

### 

In [22]:
stats_previous = df['Previous Purchases'].describe()

tabel_previous = pd.DataFrame({
    'Statistik': [
        'count',
        'mean',
        'std',
        'min',
        'Q1 (25%)',
        'Q2 (50%)',
        'Q3 (75%)',
        'max'
    ],
    'Nilai': [
        stats_previous['count'],
        stats_previous['mean'],
        stats_previous['std'],
        stats_previous['min'],
        stats_previous['25%'],
        stats_previous['50%'],
        stats_previous['75%'],
        stats_previous['max']
    ]
})

tabel_previous

,Statistik,Nilai
0,count,3900.000000
1,mean,25.351538
2,std,14.447125
3,min,1.000000
4,Q1 (25%),13.000000
5,Q2 (50%),25.000000
6,Q3 (75%),38.000000
7,max,50.000000


## Analisis Previous Purchases

Pada tahap ini, dilakukan analisis statistik deskriptif terhadap
kolom `Previous Purchases`.

Statistik yang dianalisis meliputi:
- Mean
- Standard deviation (std)
- Minimum (min)
- Maximum (max)
- Q1 atau kuartil pertama
- Q2 atau median
- Q3 atau kuartil ketiga

Fungsi `describe()` digunakan untuk memperoleh ringkasan statistik
dari kolom `Previous Purchases`.

In [23]:
stats_previous = df['Previous Purchases'].describe()

tabel_previous = pd.DataFrame({
    'Statistik': [
        'Mean',
        'Std',
        'Min',
        'Q1',
        'Q2 (Median)',
        'Q3',
        'Max'
    ],
    'Nilai': [
        stats_previous['mean'],
        stats_previous['std'],
        stats_previous['min'],
        stats_previous['25%'],
        stats_previous['50%'],
        stats_previous['75%'],
        stats_previous['max']
    ]
})

tabel_previous

,Statistik,Nilai
0,Mean,25.351538
1,Std,14.447125
2,Min,1.000000
3,Q1,13.000000
4,Q2 (Median),25.000000
5,Q3,38.000000
6,Max,50.000000


In [25]:
tabel_corr_age_purchase = pd.DataFrame({
    'Variabel 1': ['Age'],
    'Variabel 2': ['Purchase Amount (USD)'],
    'Nilai Korelasi': [corr_age_purchase]
})

tabel_corr_age_purchase

,Variabel 1,Variabel 2,Nilai Korelasi
0,Age,Purchase Amount (USD),-0.010424


## Analisis Korelasi Age vs Purchase Amount (USD)

Pada tahap ini dilakukan analisis korelasi antara usia pelanggan
(`Age`) dan jumlah pembelian (`Purchase Amount (USD)`).

Korelasi digunakan untuk mengetahui arah dan kekuatan hubungan linear
antara dua variabel.

Nilai korelasi berada pada rentang -1 sampai 1. Nilai positif
menunjukkan hubungan positif, sedangkan nilai negatif menunjukkan
hubungan negatif. Nilai yang mendekati 0 menunjukkan hubungan linear
yang lemah.

## Interpretasi Korelasi Age vs Purchase Amount (USD)

Berdasarkan tabel, nilai korelasi antara `Age` dan
`Purchase Amount (USD)` adalah -0.010424.

Nilai tersebut bernilai negatif tetapi sangat dekat dengan 0.
Hal ini menunjukkan bahwa hubungan linear antara usia pelanggan
dan jumlah pembelian bersifat negatif dan sangat lemah.

Dengan demikian, berdasarkan dataset yang digunakan, `Age` tidak
menunjukkan hubungan linear yang kuat dengan `Purchase Amount (USD)`.